sklearn/model_selection/_split.py

```diff
@@ -997,6 +997,33 @@ def split(self, X, y=None, groups=None):
             for train_index, test_index in cv.split(X, y, groups):
                 yield train_index, test_index
 
+    def get_n_splits(self, X=None, y=None, groups=None):
+        """Returns the number of splitting iterations in the cross-validator
+
+        Parameters
+        ----------
+        X : object
+            Always ignored, exists for compatibility.
```

Comment:
I don't know that much thing in the model selection base code (yet, I hope) so my comment could seems stupid, be prepared.

Is that the statement `Always ignored ...` correct. Because if I am not wrong, `cv` can be any cross-validation method, and watching around it seems that some of them need those parameters to work (e.g., `LeaveOneOut.get_n_splits` needs `X`).

@jnothman @lesteve could you confirm because if this is the case @AishwaryaRK, you will need to change the documentation accordingly.